design/rook-ceph-status-conditions.md

```diff
@@ -0,0 +1,93 @@
+  	   Status             ConditionStatus    `json:"status" description:"status of the condition, one of True, False, Unknown"`
+  	   Reason             *string            `json:"reason,omitempty" description:"one-word CamelCase reason for the condition's last transition"`
+  	   Message            *string            `json:"message,omitempty" description:"human-readable message indicating details about last transition"`
+	   LastHeartbeatTime  *unversioned.Time  `json:"lastHeartbeatTime,omitempty" description:"last time we got an update on a given condition"`
+	   LastTransitionTime *unversioned.Time  `json:"lastTransitionTime,omitempty" description:"last time the condition transition from one status to another"`
+
+The fields `Reason`, `Message`, `LastHeartbeatTime`, `LastTransitionTime` are optional field. Though the use of `Reason` field is encouraged.
+
+Condition Types field specifies the current state of the system. Condition status values may be `True`, `False`, or `Unknown`. The absence of a condition should be interpreted the same as Unknown. How controllers handle Unknown depends on the Condition in question.
+`Reason` is intended to be a one-word, CamelCase representation of the category of cause of the current status, and `Message` is intended to be a human-readable phrase or sentence, which may contain specific details of the individual occurrence. `Reason` is intended to be used in concise output, such as one-line kubectl get output, and in summarizing occurrences of causes, whereas `Message` is intended to be presented to users in detailed status explanations, such as `kubectl describe output`.
+
+In the CephClusterStatus, we can either remove the `Status.State` and `Status.Message` fields and call the `Conditions` structure from inside the `ClusterStatus`, or we can just add the `Conditions` structure by keeping the already included fields.The first method is preferred because the `Conditions` structure contains the `Conditions.Type` and `Conditions.Message` which is similar to the `Status.State` and `Status.Message`. According to the above changes, necessary changes are to be made everywhere `ClusterStatus` or one of its fields are referred. 
+
+### Examples
+Example 1
+
+
+	   Before:
+		ClusterStatus{
+		    State      :   Creating
+	   	    Message    :   The Cluster is getting created
+		    CephStatus :   *CephStatus
+		}
+	   After:
+		ClusterStatus{
+		     RookConditions{
+		          Type    :   Progressing
+			  Status  :   True
+			  Reason  :   ClusterCreating
+ 			  Message :   The Cluster is getting Created
+		     }
+		     CephStatus :   *CephStatus
+		}
+Example 2
+
+
+	   Before:
+		ClusterStatus{
+		    State      :   Updating
+	   	    Message    :   The Cluster is getting updated
+		    CephStatus :   *CephStatus
+		}
+	   After:
+		ClusterStatus{
+		     RookConditions{
+		          Type    :   Not Ready
+			  Status  :   True
+			  Reason  :   ClusterUpdating
```

Comment:
@travisn Is it ClusterUpdating or ClusterUpgrading? Or, it doesn't make any difference?

Reply:
Ideally we would differentiate between the different reasons for the operator running the reconcile loop. Right now we only really track that an orchestration is in progress and is `updating`. But even better to differentiate between updates to the cluster CR, and upgrade (because of a cluster CR update), or the operator starting it automatically when a node was added to the cluster.

Reply:
When a node is added to the cluster and rook starts reconcilation, can we consider it as Cluster Expansion?

Reply:
> But even better to differentiate between updates to the cluster CR, and upgrade (because of a cluster CR update)

I added a different ConditionType for the Cluster Upgrading and updated the document now.

Reply:
> When a node is added to the cluster and rook starts reconcilation, can we consider it as Cluster Expansion?

Adding a node increases the number of OSDs right? So isn't that the Cluster Expansion?